bots/pr/src/main/java/org/openjdk/skara/bots/pr/CheckRun.java

```diff
@@ -1136,8 +1136,36 @@ private void checkStatus() {
                 rebasePossible = false;
             }
 
-            if (rebasePossible && PullRequestUtils.isMerge(pr)) {
-                localRepo.lookup(pr.headHash()).ifPresent(this::updateMergeClean);
+            List<String> mergeJCheckMessage = new ArrayList<>();
+
+            if (PullRequestUtils.isMerge(pr)) {
+                if (rebasePossible) {
+                    localRepo.lookup(pr.headHash()).ifPresent(this::updateMergeClean);
+                }
+
+                // JCheck all commits in "Merge PR"
+                if (workItem.bot.jcheckMerge()) {
+                    var commits = localRepo.commitMetadata(localRepo.mergeBase(PullRequestUtils.targetHash(localRepo), pr.headHash()), commitHash);
+                    var commitHashes = commits.stream()
+                            .map(CommitMetadata::hash)
+                            .collect(Collectors.toSet());
+                    commitHashes.remove(pr.headHash());
+                    for (Hash hash : commitHashes) {
+                        try {
+                            PullRequestCheckIssueVisitor visitor = checkablePullRequest.createVisitor(hash);
+                            checkablePullRequest.executeChecks(hash, censusInstance, visitor, List.of(), hash);
+                            mergeJCheckMessage.addAll(visitor.messages().stream()
+                                    .map(StringBuilder::new)
+                                    .map(e -> e.append(" (failed when running jcheck with commit " + hash.hex() + ")"))
+                                    .map(StringBuilder::toString)
+                                    .toList());
+                        } catch (Exception e) {
```

Comment:
I think instead of catching Exception here, let the outer try-catch take it as that will trigger a retry and a suitable message to be printed.

Reply:
Ah, right. And I will also remove the try catch when running "second jcheck".

Reply:
Good, do that please. 

For the second jcheck, there may be a usecase for printing the exception, because the user has made edits to .jcheck/conf. On the other hand, I don't think we should ever print `e.getMessage()` on a caught `Exception` instance to the user unless we know exactly what would have thrown it. It's enough that it gets logged for now. If jcheck throws exceptions that the user needs to see, then we need to fix jcheck to present that as a valid failure instead.